File: leetcode/454.四数相加II/454.py
class Solution {
public:
    int fourSumCount(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3, vector<int>& nums4) {
        unordered_map<int, int> cntAB;
        int res = 0;
        for(const int &a:nums1){
            for(const int &b:nums2){
                cntAB[a + b]++;
            }
        }
        for(const int &c:nums3){
            for(const int &d:nums4){
                int ano = 0 - (c + d);
                if(cntAB.find(ano) != cntAB.end()){
                    res = res + cntAB[ano];
                }
            }
        }
        return res;
    }
};

File: leetcode/454.四数相加II/454.java
class Solution {
    public int fourSumCount(int[] nums1, int[] nums2, int[] nums3, int[] nums4) {
        Map<Integer, Integer> cntAB = new HashMap<>();
        int res = 0;
        for(int a:nums1){
            for(int b:nums2){
                int one = a + b;
                cntAB.put(one, cntAB.getOrDefault(one, 0) + 1); 
            }
        }
        for(int c:nums3){
            for(int d:nums4){
                int ano = 0 - (c + d);
                res = res + cntAB.getOrDefault(ano, 0);
            }
        }
        return res;
    }
}